cloudfoundry/provider/types_app.go

```diff
@@ -390,6 +418,21 @@ func mapAppValuesToType(ctx context.Context, appManifest *cfv3operation.AppManif
 	var appType AppType
 	appType.Name = types.StringValue(appManifest.Name)
 	appType.Stack = types.StringValue(appManifest.Stack)
+	// Set lifecycle from the app data if available
+	if app != nil && app.Lifecycle.Type != "" {
+		switch app.Lifecycle.Type {
+		case "buildpack":
+			appType.AppLifecycle = types.StringValue("buildpack")
+		case "docker":
+			appType.AppLifecycle = types.StringValue("docker")
+		case "cnb":
+			appType.AppLifecycle = types.StringValue("cnb")
+		default:
```

Comment:
Shouldn't the default raise an error, as only the three options from above are valid?